types/i18n/index.d.ts

```diff
@@ -450,14 +522,11 @@ declare module "i18n" {
 }
 
 declare namespace Express {
+    // tslint:disable-next-line:no-empty-interface
```

Comment:
Why declare empty interfaces?

Reply:
```
josh@Joshs-MacBook-Pro-2:$ npm run test i18n

> definitely-typed@0.0.3 test /Users/josh/Projects/OpenSource/DefinitelyTyped
> dtslint types "i18n"

Error: /Users/josh/Projects/OpenSource/DefinitelyTyped/types/i18n/i18n-tests.ts:135:26
ERROR: 135:26  expect  TypeScript@4.1 compile error: 
Property '__' does not exist on type 'Response'.
ERROR: 148:5   expect  TypeScript@4.1 compile error: 
Property '__' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 149:5   expect  TypeScript@4.1 compile error: 
Property '__' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 150:5   expect  TypeScript@4.1 compile error: 
Property '__' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 167:5   expect  TypeScript@4.1 compile error: 
Property '__n' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 168:5   expect  TypeScript@4.1 compile error: 
Property '__n' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 176:5   expect  TypeScript@4.1 compile error: 
Property '__n' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 177:5   expect  TypeScript@4.1 compile error: 
Property '__n' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 189:16  expect  TypeScript@4.1 compile error: 
No overload matches this call.
  Overload 1 of 4, '(path: PathParams, ...handlers: RequestHandler<ParamsDictionary, any, any, ParsedQs>[]): Express', gave the following error.
    Argument of type '(_req: Express.Request, res: i18n.Response) => void' is not assignable to parameter of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs>'.
      Types of parameters 'res' and 'res' are incompatible.
        Type 'Response<any, number>' is missing the following properties from type 'Response': locale, __, __n, __mf, and 5 more.
  Overload 2 of 4, '(path: PathParams, ...handlers: RequestHandlerParams<ParamsDictionary, any, any, ParsedQs>[]): Express', gave the following error.
    Argument of type '(_req: Express.Request, res: i18n.Response) => void' is not assignable to parameter of type 'RequestHandlerParams<ParamsDictionary, any, any, ParsedQs>'.
      Type '(_req: Express.Request, res: i18n.Response) => void' is not assignable to type 'RequestHandler<ParamsDictionary, any, any, ParsedQs>'.
  Overload 3 of 4, '(path: PathParams, subApplication: Application): Express', gave the following error.
    Argument of type '(_req: Express.Request, res: i18n.Response) => void' is not assignable to parameter of type 'Application'.
      Type '(_req: Request, res: Response) => void' is missing the following properties from type 'Application': init, defaultConfiguration, engine, set, and 61 more.
ERROR: 224:5   expect  TypeScript@4.1 compile error: 
Property 'setLocale' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 226:16  expect  TypeScript@4.1 compile error: 
No overload matches this call.
  Overload 1 of 4, '(path: PathParams, ...handlers: RequestHandler<ParamsDictionary, any, any, ParsedQs>[]): Express', gave the following error.
    Argument of type '(_req: Express.Request, res: i18n.Response) => void' is not assignable to parameter of type 'RequestHandler<ParamsDictionary, any, any, ParsedQs>'.
      Types of parameters 'res' and 'res' are incompatible.
        Type 'Response<any, number>' is not assignable to type 'Response'.
  Overload 2 of 4, '(path: PathParams, ...handlers: RequestHandlerParams<ParamsDictionary, any, any, ParsedQs>[]): Express', gave the following error.
    Argument of type '(_req: Express.Request, res: i18n.Response) => void' is not assignable to parameter of type 'RequestHandlerParams<ParamsDictionary, any, any, ParsedQs>'.
      Type '(_req: Express.Request, res: i18n.Response) => void' is not assignable to type 'RequestHandler<ParamsDictionary, any, any, ParsedQs>'.
  Overload 3 of 4, '(path: PathParams, subApplication: Application): Express', gave the following error.
    Argument of type '(_req: Express.Request, res: i18n.Response) => void' is not assignable to parameter of type 'Application'.
      Type '(_req: Request, res: Response) => void' is missing the following properties from type 'Application': init, defaultConfiguration, engine, set, and 61 more.
ERROR: 241:5   expect  TypeScript@4.1 compile error: 
Property 'getLocale' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 262:5   expect  TypeScript@4.1 compile error: 
Property 'getCatalog' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.
ERROR: 263:5   expect  TypeScript@4.1 compile error: 
Property 'getCatalog' does not exist on type 'Express.Request & Request<ParamsDictionary, any, any, qs.ParsedQs>'.

    at /Users/josh/Projects/OpenSource/DefinitelyTyped/node_modules/dtslint/bin/index.js:206:19
    at Generator.next (<anonymous>)
    at fulfilled (/Users/josh/Projects/OpenSource/DefinitelyTyped/node_modules/dtslint/bin/index.js:6:58)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! definitely-typed@0.0.3 test: `dtslint types "i18n"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the definitely-typed@0.0.3 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/josh/.npm/_logs/2020-12-03T23_59_19_044Z-debug.log

```
Here is what happens when I remove those "empty" interfaces. Technically they are not empty because they extend from i18nAPI. I did try to use `type` instead of interface but `type` does not work the same way. If you have another solution I will be happy to implement your idea.

Reply:
Oops, I missed the base types.  Makes sense to me.